### DISCUSSION TITLE
add support for gifs

### DISCUSSION THREAD

**Discussion by neonwatty** [neonwatty/meme-search#25] [Ideas]

**Comment by djdead** [comment]
any movement here?  lack of animated gif support is my only gripe.  and all my memes are animated. =(

**Reply by neonwatty** [reply]
Can you tell me about the speed of processing on your end? It's been my assumption that even with our smallest model at present, most users will probably experience slow image-to-text translation, and therefore converting GIFs, especially multiple frames from that GIF, would be too time-consuming for most users. At present. What's your experience?

**Reply by djdead** [reply]
I haven't really used it at all because lack of animated gif support was a dealbreaker.  I installed it, thought it was snazzy enough to use, then started uploading my images, which are mostly animated gifs.  the animated gifs weren't showing up at all.  I came on here to read the docs, saw this discussion and made my post. =) apologies for not circling back before now.  to actually address your question, though... I'm in the process of building a local LLM system using a mini PC with a 5070 nvidia gpu.  that should be able to handle any AI models I'd need.  once it arrives and I get it online, I can do some testing.

**Comment by Hp93** [comment]
Why isn't gif supported in the first place? I mean I only need a place to store and search for files. If I rename the gif extension to webp, it seems to work just fine.

**Reply by neonwatty** [reply]
Well, that's an interesting point! I didn't really think about that. Basically, what will happen at present is only one frame, probably the first, will be sent to the vision language model for image-to-text translation.

I was thinking we needed to hold off until we had a small enough model that could process multiple frames from the GIF in order to get a better description of it in total. But I suppose there's really no need to wait for that, we can just note the limitation. Because for most users, I imagine, processing just a single image is somewhat time-consuming even with our smallest models at present. And so processing a GIF, multiple frames from it would take even longer.

But there's no reason to not allow GIFs at present with that constraint clearly communicated.

**Comment by neonwatty** [comment]
Adding first-frame gif support here https://github.com/neonwatty/meme-search/pull/191

Multi-frame is definitely possible with an openai-compatible service - and those with serious local setup.

**Comment by neonwatty** [comment]
Update: a simple first version of GIF indexing has now shipped in the latest Docker images via #191. GIFs remain animated in the app, while description generation and search indexing use the first frame.
